esrally/track/loader.py

```diff
@@ -1452,17 +1465,26 @@ def parse_task(self, task_spec, ops, challenge_name, default_warmup_iterations=N
                                                      default_value=default_warmup_time_period),
                           time_period=self._r(task_spec, "time-period", error_ctx=op.name, mandatory=False,
                                               default_value=default_time_period),
+                          ramp_up_time_period=self._r(task_spec, "ramp-up-time-period", error_ctx=op.name,
+                                                      mandatory=False, default_value=default_ramp_up_time_period),
                           clients=self._r(task_spec, "clients", error_ctx=op.name, mandatory=False, default_value=1),
                           completes_parent=(task_name == completed_by_name),
                           schedule=schedule,
                           # this is to provide scheduler-specific parameters for custom schedulers.
                           params=task_spec)
         if task.warmup_iterations is not None and task.time_period is not None:
-            self._error("Operation '%s' in challenge '%s' defines '%d' warmup iterations and a time period of '%d' seconds. Please do not "
-                        "mix time periods and iterations." % (op.name, challenge_name, task.warmup_iterations, task.time_period))
+            self._error(f"Operation '{op.name}' in challenge '{challenge_name}' defines {task.warmup_iterations} "
+                        f"warmup iterations and a time period of {task.time_period} seconds but mixing time periods "
+                        f"and iterations is not allowed.")
         elif task.warmup_time_period is not None and task.iterations is not None:
-            self._error("Operation '%s' in challenge '%s' defines a warmup time period of '%d' seconds and '%d' iterations. Please do not "
-                        "mix time periods and iterations." % (op.name, challenge_name, task.warmup_time_period, task.iterations))
+            self._error(f"Operation '{op.name}' in challenge '{challenge_name}' defines a warmup time period of "
+                        f"{task.warmup_time_period} seconds and {task.iterations} iterations but mixing time periods "
+                        f"and iterations is not allowed.")
+
+        if (task.warmup_iterations is not None or task.iterations is not None) and task.ramp_up_time_period is not None:
+            self._error(f"Operation '{op.name}' in challenge '{challenge_name}' defines a ramp-up time period of "
+                        f"{task.ramp_up_time_period} seconds as well as {task.warmup_iterations} warmup iterations and "
+                        f"{task.iterations} iterations but mixing time periods and iterations is not allowed.")
```

Comment:
Re above and the fact I don't think we want to be still ramping after warmup, can we prevent and error here?

Reply:
Yes, let's prevent this. I've pushed 518d66c0bea9cb7b8c77ddd92ba1b25fb678a46d to enforce this behavior.